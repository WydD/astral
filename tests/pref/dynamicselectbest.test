<?xml version="1.0" encoding="UTF-8"?>
<astral xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://astral.ligforge.imag.fr/schema/query.xsd">
	<query t0="auto">
		<selectbest>
			<preference preferred="YEARS==90" over="YEARS=80" ignored="TITLE,DIRECTOR,ACTOR"/>
			<preference condition="DIRECTOR=='Woody Allen'" preferred="ACTOR=='Charlotte Rampling'" over="ACTOR=='Mia Farrow'" ignored="TITLE"/>
			<preference condition="GENRE=='comedy' &amp;&amp; YEARS==80" preferred="DIRECTOR=='Joel Coen'" over="DIRECTOR=='Woody Allen'" ignored="TITLE,ACTOR"/>
			<preference condition="YEARS==80" preferred="GENRE=='drama'" over="GENRE=='comedy'" ignored="TITLE"/>
			<pi removeAttributes="T">
			<window>
				<description wtype="P" a="1" b="-5" c="1" d="0" rate="1"/> 
				<source id="S"/>
			</window>
			</pi>
		</selectbest>
	</query>
	<sources>
		<source file="tests/pref/S.S" id="S" impl="FileStreamImpl"/>
	</sources>
</astral>
