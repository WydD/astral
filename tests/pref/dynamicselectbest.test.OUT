DIRECTOR,YEARS,GENRE,ACTOR,TITLE
t=1
James Cameron,90,drama,Bill Paxton,Titanic
t=5
James Cameron,90,drama,Bill Paxton,Titanic
Joel Coen,80,comedy,Louise Lasser,Crimewave
t=6
James Cameron,90,drama,Bill Paxton,Titanic
Joel Coen,80,comedy,Louise Lasser,Crimewave
James Cameron,0,action,Zoe Saldana,Avatar
